parsers/CZ.py

```diff
@@ -0,0 +1,221 @@
+    xml = BeautifulSoup(content, "xml")
+    mapper = get_mapper(xml)
+
+    data_tag = xml.find("data")
+    data_list = []
+
+    for values in data_tag:
+        data = {
+            "zoneKey": zone_key,
+            "production": {},
+            "storage": {},
+            "source": url,
+            "datetime": datetime.fromisoformat(values["date"]).isoformat(),
+        }
+
+        for k, v in mapper.items():
+            generator = translate_table_gen[k]
+            if k != "PsPP":
+                data["production"][generator] = float(values[v])
+            else:
+                data["storage"][generator] = float(values[v])
+
+        data_list.append(data)
+
+    return data_list
+
+
+def fetch_exchange(
+    zone_key1: str = "CZ",
+    zone_key2: str = "DE",
+    session: Optional[Session] = None,
+    target_datetime: Optional[datetime] = None,
+    logger: Logger = getLogger(__name__),
+    mode: Optional[str] = "Actual",
+):
+    if not target_datetime:
+        target_datetime = datetime.now().replace(minute=0, second=0, microsecond=0)
+    from_datetime = target_datetime - timedelta(hours=1)
+
+    payload = """<?xml version="1.0" encoding="utf-8"?>
+<soap12:Envelope xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:soap12="http://www.w3.org/2003/05/soap-envelope">
+  <soap12:Body>
+    <CrossborderPowerFlows xmlns="https://www.ceps.cz/CepsData/">
+      <dateFrom>{0}</dateFrom>
+      <dateTo>{1}</dateTo>
+      <agregation>{2}</agregation>
+      <function>{3}</function>
+      <version>{4}</version>
+    </CrossborderPowerFlows>
+  </soap12:Body>
+</soap12:Envelope>""".format(
+        from_datetime.isoformat(), target_datetime.isoformat(), "QH", "AVG", "RT"
+    )
+
+    content = make_request(session, payload, zone_key1).content
+    xml = BeautifulSoup(content, "xml")
+    mapper = get_mapper(xml)
+
+    data_tag = xml.find("data")
+    data_list = []
+
+    for values in data_tag:
+        data = {
+            "sortedZoneKeys": f"{zone_key1}->{zone_key2}",
+            "datetime": datetime.fromisoformat(values["date"]).isoformat(),
+            "netFlow": 0.0,
+            "source": url,
+        }
+
+        for k, v in mapper.items():
+            country = "".join(
+                [c for key, c in translate_table_dist.items() if key in k and mode in k]
+            )
+            if country != "" and country in (zone_key1, zone_key2):
+                data["netFlow"] += float(values[v])
+
+        data_list.append(data)
+
+    return data_list
+
+
+def fetch_exchange_forecast():
+    fetch_exchange("CZ", "DE", mode="Planned")
+
+
+def fetch_price(
+    zone_key: str = "CZ",
+    session: Optional[Session] = None,
+    target_datetime: Optional[datetime] = None,
+    logger: Logger = getLogger(__name__),
+):
+
+    if not target_datetime:
+        target_datetime = arrow.now().replace(minute=0, second=0)
+    from_datetime = target_datetime.shift(days=-5)
+
```

Comment:
This still use arrow, if we switch this to datetime like the others when we can remove the arrow import.